File: src/main/java/fr/traqueur/energylib/commands/DeleteCommand.java
package fr.traqueur.energylib.commands;

import fr.traqueur.commands.api.Arguments;
import fr.traqueur.commands.api.Command;
import fr.traqueur.energylib.EnergyLib;
import fr.traqueur.energylib.api.EnergyManager;
import fr.traqueur.energylib.api.components.EnergyNetwork;
import org.bukkit.command.CommandSender;

public class DeleteCommand extends Command<EnergyLib> {

    private final EnergyManager manager;

    public DeleteCommand(EnergyLib plugin) {
        super(plugin, "delete");
        this.manager = plugin.getManager();
        this.setPermission("energy.admin.delete");
        this.setUsage("/energy-admin delete <network>");
        this.setDescription("Delete a network.");
        this.addArgs("network:network");
    }

    @Override
    public void execute(CommandSender commandSender, Arguments arguments) {
        EnergyNetwork network = arguments.get("network");
        manager.deleteNetwork(network);
        commandSender.sendMessage("§aThe network §e" + network.getId() + " §ain chunk §e" + network.getChunk().getX() + " " + network.getChunk().getZ() + " §ahas been deleted.");
    }
}

File: src/main/java/fr/traqueur/energylib/commands/DeleteCommand.kt
package fr.traqueur.energylib.commands

import fr.traqueur.commands.api.Arguments
import fr.traqueur.commands.api.Command
import fr.traqueur.energylib.EnergyLib
import fr.traqueur.energylib.api.EnergyManager
import fr.traqueur.energylib.api.components.EnergyNetwork
import org.bukkit.command.CommandSender

class DeleteCommand(plugin: EnergyLib) : Command<EnergyLib?>(plugin, "delete") {
    private val manager: EnergyManager? = plugin.manager

    init {
        this.permission = "energy.admin.delete"
        this.usage = "/energy-admin delete <network>"
        this.description = "Delete a network."
        this.addArgs("network:network")
    }

    override fun execute(commandSender: CommandSender, arguments: Arguments) {
        val network = arguments.get<EnergyNetwork>("network")
        manager?.deleteNetwork(network)
        commandSender.sendMessage(
            "§aThe network §e" + network.id + " §ain chunk §e" + network.chunk
                ?.x + " " + network.chunk?.z + " §ahas been deleted."
        )
    }
}
